src/plugins/intel_cpu/src/cpu_shape.h

```diff
@@ -13,15 +13,44 @@
 
 namespace MKLDNNPlugin {
 
+class Interval {
+public:
+    Interval(Dim val) {
+        minValue = val;
+        maxValue = val;
+    }
+
+    Interval(Dim minVal, Dim maxVal) {
+        minValue = minVal;
+        maxValue = maxVal;
+    }
+
+    bool isStatic() const {
```

Comment:
I guess that `isStatic` more understandable
but what do you think about renaming `Interval` -> `CpuDim`?

Reply:
As long as we call this entity an Interval, we have to use definitions appropriate for intervals, and in that paradigm interval [a,a] is called a degenerate interval. `CpuDim` is rather confusing since we already have `MKLDNNPlugin::Dim`. I think we need to make a fundamental decision about how we handle particular dims in dynamic shapes. On the other hand, one step further and we will have the same shape description as the ngraph::partial_shape.

Reply:
> On the other hand, one step further and we will have the same shape description as the ngraph::partial_shape.

Yes, here I noted https://github.com/openvinotoolkit/openvino/pull/9380#issuecomment-1000829520 that in most cases for plug-in enough to know whether dim static or not.

I suggested `CpuDim` only because we already have MKLDNNPlugin::Dim, which just `using` for `size_t`

Reply:
I would say that it is not that simple. When we introduce these intervals, we significantly extend the `Shape` interface and, which is the most important thing, add a duplication for the same actions. For example, using the previous interface the only logical way to check if the particular axis is undefined was to get the dimensions and check the special value. But now, we can get dimensions as it was before, or, get the intervals and check them. And the same picture for all other actions. We can work either through getIntervals or getDims mostly equally. On the contrary, ngraph::partial_shape does not provide such freedom allowing work only with intervals.

Reply:
We can leave only ctor `Shape(std::vector<Interval>)` and create Interval from minDims, maxDims

Reply:
BTW, another approach could be to add a setter to the shape class, so we can change a particular dimmention in the shape. M.b. something like that `setDim(size_t idx, Dim minDim, Dim maxDim)`

Reply:
I'd like to leave `Shape` immutable

Reply:
> We can leave only ctor `Shape(std::vector<Interval>)`, remove `getInervalDims` and create `Interval` from `minDims`, `maxDims`

Following this logic, we can just add a constructor from two min/max Dims vectors `Shape(minDims, maxDims)` or from a vector of min max pairs without introduction of the Interval class.

Reply:
> add a constructor from two min/max Dims vectors Shape(minDims, maxDims)
Sounds good!

Reply:
Since this changes initially was introduced in [PR ](https://github.com/openvinotoolkit/openvino/pull/9002)
@nshchego Do you have any objections?

Reply:
Intervals make current Shape implementation more flexible. They allow to:
1. Operates with separate dimensions. That is useful when dimensions are handled separately and then used to create new shape.
2. Make easy initialization like Shape{ {min1, max1}, scalar2, inteval3 }

Reply:
Then we need to decide whether we use the `Interval` to handle dynamic dimensions  or the current interface. Both of them together will give us to much flexibility, as I've mentioned before. Moreover, `Shape` was designed as a lightweight, in a sense of performance, alternative to `ngraph::partial_shape`, but now we are trying to transform it to the same abstraction, which is not a wrong direction in itself, but we need to consider the performance impacts. Specifically, `getInervalDims` always rebuilds the intervals vector, the interface users may (and with no doubts will) use the intervals where the dims just enough, impacting performance with extra range calculation. The first one can be addressed by storing the vector of intervals directly, the second is more complicated and requires some optimizations on the range computation (caching?) or at least an estimation of possible penalty from transition from dims to the intervals.

Reply:
At this moment we need to work with separated dimension in 4 - 5 places in plug-in. From my point of view, in this case enough to use this approach: https://github.com/openvinotoolkit/openvino/pull/9380#discussion_r775029109.
With this ctor `Shape(minDims, maxDims)` we still have flexible `Shape` implementation and can create `Shape` without `PartialShape`, but we avoid situation with two ways getting dims: https://github.com/openvinotoolkit/openvino/pull/9380#discussion_r775027206

As described here: https://github.com/openvinotoolkit/openvino/pull/9380#discussion_r775384811 if we introduce `Interval` we will have to cache `std::vector<Interval>` and we will store in `Shape` dims, minDim, maxDims and std::vector<Interval>which are interchangeable and storing all of this data looks excessively.

Since most part of plug-in use approach with `getDims(), getMaxDims(), getMinDims()` we can't migrate from dims to Intervals.
So I guess better to use approach with `Shape(minDims, maxDims)`